### PR TITLE
oraasm: new resource agent for Oracle ASM Disk Groups

### PR DESCRIPTION
**Pull request by oalbrigt**

### CODE REVIEW

File: heartbeat/oraasm [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+oraasm_status() {
+	rc=$(oraasm_monitor)
+	return $rc
+}
+
+oraasm_validate_all() {
+	if [ x = "x$OCF_RESKEY_home" ]; then
+		ocf_exit_reason "home not set"
+		return $OCF_ERR_INSTALLED
```
Comment (dmuhamedagic) [inline heartbeat/oraasm]: This should be OCF_ERR_CONFIGURED.

File: heartbeat/oraasm [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+	validate-all
+	methods
+	meta-data
+	usage
+	!
+}
+
+oraasm_getconfig() {
+    [ x = "x$OCF_RESKEY_home" ] &&
+        OCF_RESKEY_home=`awk -F: "/^+$OCF_RESKEY_diskgroup:/"'{print $2}' /etc/oratab`
```
Comment (dmuhamedagic) [inline heartbeat/oraasm]: Looks like you have indentation with spaces here.

File: heartbeat/oraasm [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        return $OCF_SUCCESS
+    fi
+
+    ocf_run -q /etc/init.d/ohasd start
+
+    while ! oraasm_monitor; do
+        sleep 1
+    done
+
+    if [ $? -eq 0 ]; then
```
Comment (dmuhamedagic) [inline heartbeat/oraasm]: This test is superfluous. You won't get here unless the monitor succeeds.

**Comment by oalbrigt** [comment]
Updated based on feedback from @dmuhamedagic.

Yeah, I did check how ohasd works.

File: heartbeat/oraasm [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+<content type="string" default="" />
+</parameter>
+
+</parameters>
+
+<actions>
+<action name="start" timeout="120" />
+<action name="stop" timeout="120" />
+<action name="status" timeout="60" />
+<action name="monitor" depth="0" timeout="30" interval="10" start-delay="10" />
```
Comment (dmuhamedagic) [inline heartbeat/oraasm]: Forgot to take a look at the operation timeouts. The timeouts are the minimum recommended values for some typical setup. Put differently, they should be good for most deployments. But they should not be too big either. Are these timeouts just a copy from another RA?

Comment (dmuhamedagic) [inline heartbeat/oraasm]: Also, why "start-delay"?

Comment (oalbrigt) [inline heartbeat/oraasm]: Yeah. They are copied from another agent, so I'll update them to 60 for start/stop and 30 for status/monitor and remove the start-delay.

**Pull request merged**